src/login-hooks.jsx

```diff
@@ -0,0 +1,17 @@
+import { createContext, useContext, useState } from "react";
+
+const LoginContext = createContext(false);
+
+export function LoginProvider({ children }) {
+  const [isLogin, setIsLogin] = useState(false);
+  const switchLoginState = () => setIsLogin(!isLogin);
+
+  return (
+    <LoginContext.Provider value={[isLogin, switchLoginState]}>
```

Comment:
[q]
```suggestion
    <LoginContext.Provider value={{ isLogin, switchLoginState }}>
```

value を配列にしているって何か理由ありますでしょうか！
単にオブジェクトで良いかなと思い質問です！

Reply:
複数の値を渡す際に配列形式```[isLogin, switchLoginState]```を使ったのは、
React の ```useState``` フックなどで配列を使用していたため（```const [state, setState] = useState(...)``` のような書き方に慣れていたため）、無意識にその形式を選択していたのだと思います。

オブジェクトであれば、順序に関係なくプロパティ名でアクセスでき、可読性・保守性が高くなると思い、オブジェクトに変更いたしました。
コミット：